java/src/jmri/jmrit/logixng/actions/AnalogActionLightIntensity.java

```diff
@@ -0,0 +1,289 @@
+    }
+
+    public void setLight(@Nonnull VariableLight light) {
+        assertListenersAreNotRegistered(log, "setLight");
+        setLight(InstanceManager.getDefault(NamedBeanHandleManager.class)
+                .getNamedBeanHandle(light.getDisplayName(), light));
+    }
+
+    public void removeLight() {
+        assertListenersAreNotRegistered(log, "setLight");
+        if (_lightHandle != null) {
+            InstanceManager.lightManagerInstance().removeVetoableChangeListener(this);
+            _lightHandle = null;
+        }
+    }
+
+    public NamedBeanHandle<VariableLight> getLight() {
+        return _lightHandle;
+    }
+
+    public void setAddressing(NamedBeanAddressing addressing) throws ParserException {
+        _addressing = addressing;
+        parseFormula();
+    }
+
+    public NamedBeanAddressing getAddressing() {
+        return _addressing;
+    }
+
+    public void setReference(@Nonnull String reference) {
+        if ((! reference.isEmpty()) && (! ReferenceUtil.isReference(reference))) {
+            throw new IllegalArgumentException("The reference \"" + reference + "\" is not a valid reference");
+        }
+        _reference = reference;
+    }
+
+    public String getReference() {
+        return _reference;
+    }
+
+    public void setLocalVariable(@Nonnull String localVariable) {
+        _localVariable = localVariable;
+    }
+
+    public String getLocalVariable() {
+        return _localVariable;
+    }
+
+    public void setFormula(@Nonnull String formula) throws ParserException {
+        _formula = formula;
+        parseFormula();
+    }
+
+    public String getFormula() {
+        return _formula;
+    }
+
+    private void parseFormula() throws ParserException {
+        if (_addressing == NamedBeanAddressing.Formula) {
+            Map<String, Variable> variables = new HashMap<>();
+
+            RecursiveDescentParser parser = new RecursiveDescentParser(variables);
+            _expressionNode = parser.parseExpression(_formula);
+        } else {
+            _expressionNode = null;
+        }
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public Category getCategory() {
+        return Category.ITEM;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public boolean isExternal() {
+        return true;
+    }
+    
+    @Override
+    public void vetoableChange(java.beans.PropertyChangeEvent evt) throws java.beans.PropertyVetoException {
+        if ("CanDelete".equals(evt.getPropertyName())) { // No I18N
+            if (evt.getOldValue() instanceof VariableLight) {
+                if (evt.getOldValue().equals(getLight().getBean())) {
+                    PropertyChangeEvent e = new PropertyChangeEvent(this, "DoNotDelete", null, null);
+                    throw new PropertyVetoException(Bundle.getMessage("LightIntensity_LightInUseVeto", getDisplayName()), e); // NOI18N
+                }
+            }
+        } else if ("DoDelete".equals(evt.getPropertyName())) { // No I18N
```

Comment:
The DoDelete portion of the veto is not needed when the CanDelete reply is DoNotDelete.

Veto process fails for this change and the previous Light intensity action:
```
     [java] java.util.concurrent.ExecutionException: java.util.MissingResourceException: Resource 'LightIntensity_LightInUseVeto' not found
```

Reply:
Thanks. Fixed for both ActionLightIntensity and AnalogActionLightIntensity.